cloudbuild.yaml

```diff
@@ -47,209 +47,98 @@
 #    https://github.com/GoogleCloudPlatform/cloud-builders/tree/master/go
 
 steps:
-- id: 'Docker Image: open-match-base-build'
+- id: 'Docker Image: open-match-build'
   name: gcr.io/cloud-builders/docker
-  args: ['build', '-t', 'open-match-base-build', '-f', 'Dockerfile.base-build', '.']
+  args: ['build', '--build-arg', 'GCP_PROJECT_ID=$PROJECT_ID', '-t', 'open-match-build', '-f', 'Dockerfile', '.']
   waitFor: ['-']
 
-- id: 'Docker Image: backendapi'
-  name: gcr.io/cloud-builders/docker
-  args: ['build', '-t', 'gcr.io/$PROJECT_ID/openmatch-backendapi:${_OM_VERSION}-${SHORT_SHA}', 'cmd/backendapi']
-  waitFor: ['Docker Image: open-match-base-build']
-
-- id: 'Docker Image: frontendapi'
-  name: gcr.io/cloud-builders/docker
-  args: ['build', '-t', 'gcr.io/$PROJECT_ID/openmatch-frontendapi:${_OM_VERSION}-${SHORT_SHA}', 'cmd/frontendapi']
-  waitFor: ['Docker Image: open-match-base-build']
-
-- id: 'Docker Image: mmforc'
-  name: gcr.io/cloud-builders/docker
-  args: ['build', '-t', 'gcr.io/$PROJECT_ID/openmatch-mmforc:${_OM_VERSION}-${SHORT_SHA}', 'cmd/mmforc']
-  waitFor: ['Docker Image: open-match-base-build']
-
-- id: 'Docker Image: mmlogicapi'
-  name: gcr.io/cloud-builders/docker
-  args: ['build', '-t', 'gcr.io/$PROJECT_ID/openmatch-mmlogicapi:${_OM_VERSION}-${SHORT_SHA}', 'cmd/mmlogicapi']
-  waitFor: ['Docker Image: open-match-base-build']
-
-- id: 'Docker Image: Evaluator Simple'
-  name: gcr.io/cloud-builders/docker
-  args: ['build', '-t', 'gcr.io/$PROJECT_ID/openmatch-evaluator-simple:${_OM_VERSION}-${SHORT_SHA}', 'examples/evaluators/golang/simple']
-  waitFor: ['Docker Image: open-match-base-build']
-
-- id: 'Docker Image: openmatch-mmf-cs-mmlogic-simple'
-  name: gcr.io/cloud-builders/docker
-  args: ['build', '-t', 'gcr.io/$PROJECT_ID/openmatch-mmf-cs-mmlogic-simple:${_OM_VERSION}-${SHORT_SHA}', '.']
-  dir: 'examples/functions/csharp/simple'
-  waitFor: ['Docker Image: open-match-base-build']
-
-- id: 'Docker Image: openmatch-mmf-go-mmlogic-simple'
-  name: gcr.io/cloud-builders/docker
-  args: ['build', '-t', 'gcr.io/$PROJECT_ID/openmatch-mmf-go-mmlogic-simple:${_OM_VERSION}-${SHORT_SHA}', '-f', 'examples/functions/golang/manual-simple/Dockerfile', '.']
-  waitFor: ['Docker Image: open-match-base-build']
+- id: 'Build: Clean'
+  name: open-match-build
+  args: ['make', 'clean']
+  waitFor: ['Docker Image: open-match-build']
```

Comment:
Would it make sense to push the open-match-build image to gcr and use it directly instead of building the base environment image per PR? Or we can use [Kaniko cache](https://cloud.google.com/cloud-build/docs/kaniko-cache) to rebuild the base image with expiration time in case any build time dependencies have an update.

Reply:
Kaniko cache looks like it would make much more sense here.

Reply:
I hit https://github.com/GoogleContainerTools/kaniko/issues/477 immediately.

Reply:
I think it would be ideal to have a caching system for the CI base image. I tried to build just the base image on my workstation yesterday and it took me ~10.5 minutes.

Reply:
I removed the C++ build stuff just now that should improve it significantly.